txnbuild/helpers.go

```diff
@@ -192,3 +192,20 @@ func NewValidationError(field, message string) *ValidationError {
 		Message: message,
 	}
 }
+
+// Parses an asset string in canonical form (SEP-11) into an Asset structure.
+// https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0011.md#asset
+func ParseAssetString(canonical string) (Asset, error) {
+	assets, err := xdr.BuildAssets(canonical)
+	if err != nil {
+		return &NativeAsset{}, errors.Wrap(err, "failed parsing: is it in canonical (SEP-11) form?")
```

Comment:
I'd keep the error specific to what we know as facts.
```suggestion
		return &NativeAsset{}, errors.Wrap(err, "error parsing asset string")
```